### PR TITLE
Update Azure HDInsight workflows

### PR DESCRIPTION
**Pull request by nebilbenmabrouk**
Two workflows are updated: 

1. BigData/resources/catalog/Azure_HDInsight_Create_Spark_Cluster.xml
2. BigData/resources/catalog/Azure_HDInsight_Spark_Submit_Application.xml

### CODE REVIEW

File: BigData/resources/catalog/Azure_HDInsight_Spark_Submit_Application.xml [hunk 1]
```diff
@@ -1,101 +1,53 @@
...
-    # copy the script to the remote machine
-    sshpass -p $sshPassword scp -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -P $sshPort scripts/script.sh $sshUser@$sshHost:/tmp/script.sh
-    
-    # execute the script in the remote machine
-    sshpass -p $sshPassword ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no $sshUser@$sshHost -p $sshPort "/tmp/script.sh"
-fi
+mkdir -p /tmp/${jobId}/spark-application
+cd /tmp/${jobId}/spark-application
+wget -O spark-application-${jobId}.jar -nc ${appURL}
+spark-submit --class ${appClass} --master yarn spark-application-${jobId}.jar
```
[the remark this answers is not in the document]
Comment (fviale) [reply BigData/resources/catalog/Azure_HDInsight_Spark_Submit_Application.xml:40]: you can use def jsonSlurper = new JsonSlurper() to parse json variables in groovy, but that would require to change to groovy your script and use "command ...".execute().text to run bash commands

[the remark this answers is not in the document]
Comment (nebilbenmabrouk) [reply BigData/resources/catalog/Azure_HDInsight_Spark_Submit_Application.xml:40]: The argument handling is already done. 
In the top of the task implementation there ara these lines: 
```
jobId=$variables_PA_JOB_ID
appURL=$variables_SPARK_APPLICATION_URL
appClass=$variables_SPARK_APPLICATION_CLASS
```

[the remark this answers is not in the document]
Comment (fviale) [reply BigData/resources/catalog/Azure_HDInsight_Spark_Submit_Application.xml:40]: I was talking about possible parameters of the spark application

[the remark this answers is not in the document]
Comment (fviale) [reply BigData/resources/catalog/Azure_HDInsight_Spark_Submit_Application.xml:40]: `./bin/spark-submit \
  --class <main-class> \
  --master <master-url> \
  --deploy-mode <deploy-mode> \
  --conf <key>=<value> \
  ... # other options
  <application-jar> \
  [application-arguments]`

==> application-arguments

[the remark this answers is not in the document]
Comment (fviale) [reply BigData/resources/catalog/Azure_HDInsight_Spark_Submit_Application.xml:40]: alternatively, you could simply add a variable that contain application-arguments as a string (default to empty)

[the remark this answers is not in the document]
Comment (nebilbenmabrouk) [reply BigData/resources/catalog/Azure_HDInsight_Spark_Submit_Application.xml]: This is what I did. I added two separate variables:

1. Submission parameters (e.g., --class org.apache.spark.examples.SparkPi --master yarn)
2. Application arguments

Thanks Fabien.

[the remark this answers is not in the document]
Comment (fviale) [reply BigData/resources/catalog/Azure_HDInsight_Spark_Submit_Application.xml]: thanks @nebilbenmabrouk 

could you please add a correct description to the variables you just added

[the remark this answers is not in the document]
Comment (nebilbenmabrouk) [reply BigData/resources/catalog/Azure_HDInsight_Spark_Submit_Application.xml]: Done. Thanks again.

**Review by fviale — APPROVED** [review]
LGTM

**Pull request merged**